### ISSUE TITLE
Upgrade Typescript to v4.3

### ISSUE THREAD

**Issue by topvis** [quasarframework/quasar#10388]
**Is your feature request related to a problem? Please describe.**
Currently the typescript version used by Quasar is 4.2.2.
Typescript 4.3 was released in May. The particular feature I need is it allows private methods (with # prefix) in Classes.

**Describe the solution you'd like**
Upgrade to TS4.3

**Describe alternatives you've considered**
NA

**Additional context**
It's actually a question. If I upgrade typescript by NPM or Yarn for my project by myself, will it break anything? Or I have to wait for Quasar to upgrade it?

**Comment by IlCallo** [comment]
If you install TS 4.3 into your project, Quasar should pick up your version instead of the one we provide from `@quasar/app`
Take into consideration that we upgrade deps only after testing them out, if you install a new version (especially for TS which doesn't follow semver) expect some bumps and errors we cannot help you with

**Comment by razorness** [comment]
Version bump would be really helpful. See: https://github.com/vuejs/vue-next/issues/4307
I faced this issue after upgrading to `@quasar/app v3.1.0` which introduces `vue v3.2.4`.

The current solution is to set a resultion in `package.json`:

```
{
	// ...
	"resolutions": {
		"typescript": "^4.3.5"
	},
	// ...
}
```

**Comment by ovidiuc** [comment]
> If you install TS 4.3 into your project, Quasar should pick up your version instead of the one we provide from `@quasar/app` Take into consideration that we upgrade deps only after testing them out, if you install a new version (especially for TS which doesn't follow semver) expect some bumps and errors we cannot help you with

@IlCallo from my experience, that is not reliable at the moment. When you first run "quasar dev" the Quasar installed version is used (currently 4.2.2). After you modify `tsconfig.json` it seems it picks up the separately installed version (for me 4.4.3). So somehow related to hot reloading. I can submit a bug report in a separate issue if this is supposed to work, let me know.

**Comment by IlCallo** [comment]
Closing as TS has been upgraded to 4.4.2 into latest release of `@quasar/app`

@ovidiuc that's strange, and the first time someone reports this behaviour. I just tested this and cannot reproduce it, you probably have something into your project forcing this behaviour

**Comment by ovidiuc** [comment]
@IlCallo I've tested it in an isolated environment with a clean starter project and I was able to reproduce the behavior. With the upgrade to 4.4.2 I'm unable to test it, as there are no different TS configuration options between 4.4.3 and 4.4.2. 

Here are the steps to reproduce for the previous version, maybe someone else gets into the same issue:
1. Generate a new project
```
$ quasar create v2ts

  ___
 / _ \ _   _  __ _ ___  __ _ _ __
| | | | | | |/ _` / __|/ _` | '__|
| |_| | |_| | (_| \__ \ (_| | |
 \__\_\\__,_|\__,_|___/\__,_|_|



? Project name (internal usage for dev) v2ts
? Project product name (must start with letter if building mobile apps) Quasar App
? Project description A Quasar Framework app
? Author
? Pick your CSS preprocessor: SCSS
? Check the features needed for your project: ESLint (recommended), TypeScript, Vuex, Axios, Vue-i18n
? Pick a component style: Composition
? Pick an ESLint preset: Standard
? Continue to install project dependencies after the project has been created? (recommended) yarn

  Quasar CLI · Generated "v2ts".
```
2. Downgrade quasar app:
```
yarn add --dev @quasar/app@3.1.0
```
3. Install typescript 4.4.3 and check to see both version installed (4.2.2 and 4.4.3):
```
yarn add --dev typescript@4.4.3
yarn why typescript
```
4. Add `"useUnknownInCatchVariables": false` to `tsconfig.json`:
```
{
  "extends": "@quasar/app/tsconfig-preset",
  "compilerOptions": {
    "baseUrl": ".",
    "useUnknownInCatchVariables": false
  }
}
```
5. Run `quasar dev`.

```
 App •  ERROR  •  UI  in ./src/App.vue?vue&type=script&lang=ts

Error: error while parsing tsconfig.json




 App •  COMPILATION FAILED  • Please check the log above for details.
```

6. With the server running, delete `"useUnknownInCatchVariables": false` from `tsconfig.json`. Save.
Check the server: no errors.
7. With the server still running add back `"useUnknownInCatchVariables": false`. Save. 
Check the server: no errors.
